lib/Module/IntrinsicCleaner.cpp

```diff
@@ -49,6 +49,54 @@ bool IntrinsicCleanerPass::runOnModule(Module &M) {
   return dirty;
 }
 
+llvm::Constant* IntrinsicCleanerPass::createConstantVector(llvm::LLVMContext& ctx, 
+  llvm::Value& constantVector, opMode mode) { 
+  llvm::Constant* pConstant = nullptr;
+  llvm::ConstantInt* pCInt = nullptr;
+  Type* pType = nullptr;
+  Type* elementType = nullptr;
+  unsigned elementNum = 0;
+  unsigned elementBw = 0;
+  SmallVector<llvm::Constant*, 8> CElts;
+
+  pType = constantVector.getType();
+  if (!pType) { 
```

Comment:
Write defensive here and in similar places.
You always expect that `getType()` returns a non-`nullptr` value.
Check this with an `assert()`, i.e. `assert(pType && "Couldn't resolve type");`.
Instead, if you return `nullptr` you ignore this issue and your caller has to handle this.
In this case, if `getType()` returns `nullptr` than probably LLVM's API has changed.